### ISSUE TITLE
segment ids worng?

### ISSUE THREAD

**Issue by jjymhkx0820** [chqiwang/gpsd-rec#2]
segment_ids = torch.zeros_like(input_item_ids)
segment_ids[:, historical_len] = 1

historical_len is a [bath] shape tensor, this code make every record have same segment_ids.

I think segment means target sign embdding? 

maybe segment_ids[:, historical_len.unsqueeze(1)] = 1 is right?

**Comment by chqiwang** [comment]
You're right! That's a coding mistake! 
Did you observe any performance difference after fixing it?

**Comment by jjymhkx0820** [comment]
sorry，I just debugged the model and didn’t run any dataset for performance observe...

**Comment by chqiwang** [comment]
Similar performances were observed after fixing the segment id.
<img width="1415" height="611" alt="Image" src="https://github.com/user-attachments/assets/f0bd0cc0-9338-4dd4-9446-e4f21710fb38" />

**Comment by jjymhkx0820** [comment]
So, distinguishing whether it is a “final click label” or an “infer item” is useless, no matter in AR tasks or discrimination tasks, right? . . .

**Comment by chqiwang** [comment]
Even without segment id, the network can also distinguish them by checking "whether it is the last token".

**Comment by jjymhkx0820** [comment]
Interesting, but it seems difficult to clearly distinguish "whether it is the last token" in the current input. The last token doesn't seem to have a direct length signal. What is the purpose of the mask?

**Comment by chqiwang** [comment]
The Transformer part may have no idea which is the last token. However, the ranking head is built on the top of the last token. Therefore the model is able to identify which token is the last token.

**Comment by tradingtransformer** [comment]
I find no difference when I removed segment id, maybe it is not needed? btw, when will the other datasets become available?

**Comment by chqiwang** [comment]
> I find no difference when I removed segment id, maybe it is not needed? btw, when will the other datasets become available?

We've finished the code for the other two datasets. Thank you for your attention.
